test/integration_test/basic_test.go

```diff
@@ -609,3 +617,94 @@ func BasicPvcControllerRegression(tc *types.TestCase) func(*testing.T) {
 		ci_utils.UninstallAndValidateStorageCluster(cluster, t)
 	}
 }
+
+// AlertManagerRegression test includes the following steps:
+// 1. Deploy PX and validate AlertManager is not enabled by default
+// 2. Create ALertManager secret
+// 3. Enable AlertManager without Prometheus and validate it is not getting deployed
+// 4. Enable AlertManaget with Prometheus and validate it gets deployed
+// 5. Delete AlertManager pods and validate they get re-deployed
+// 6. Disable AlertManager in StorageCluster and validate components
+// 7. Delete AlertManager secret
+// 8. Delete StorageCluster and validate it got successfully removed
+func AlertManagerRegression(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		var err error
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
+
+		// Create and validate StorageCluster
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Validate AlertManager is not enabled by default
+		logrus.Info("Validate ALertManager is not enabled by default")
+		if cluster.Spec.Monitoring != nil {
+			if cluster.Spec.Monitoring.Prometheus != nil {
+				if cluster.Spec.Monitoring.Prometheus.Enabled {
```

Comment:
Don't need to check this to see if alert manager is enabled by default or not.

Reply:
Done